pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BasePartitionUpsertMetadataManager.java

```diff
@@ -88,6 +88,8 @@
 @ThreadSafe
 public abstract class BasePartitionUpsertMetadataManager implements PartitionUpsertMetadataManager {
   protected static final long OUT_OF_ORDER_EVENT_MIN_REPORT_INTERVAL_NS = TimeUnit.MINUTES.toNanos(1);
+  // The special value to indicate the largest comparison value is not set yet.
+  private static final double LARGEST_COMPARISON_VALUE_NOT_SET = Double.MIN_VALUE;
```

Comment:
if the idea is to support negative comparison value then i think you might want to use `NEGATIVE_INFINITY` instead

Reply:
good catch. you're right.